pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -658,19 +673,20 @@ func (r *ReconcileComplianceScan) phaseAggregatingHandler(h scanTypeHandler, log
 		return reconcile.Result{}, err
 	}
 
-	instanceCopy := instance.DeepCopy()
-
-	if instanceCopy.Annotations == nil {
-		instanceCopy.Annotations = make(map[string]string)
+	// check if instance has any annotation before checking for check count
+	// to avoid nil pointer exception in case of missing annotation
```

Comment:
Was this a bug prior to the CEL work, too? If so - couldn't we pull this into a separate PR?

Reply:
yes I think that makes sense